js/msp/msp_helperfunctions.js

```diff
@@ -0,0 +1,1813 @@
+'use strict';
+MSP.listen(MSP_process_data);
+function MSP_process_data (dataHandler) {
```

Comment:
Yes, that is more what I had in mind. This is just the first part, and this PR was more to isolate the protocol code so the rest could be further improved without the need to change that. 
I think process_data and also MSP.crunch() method didn't anyway use the benefit of being part of the same object as the protocol since they were mostly static functions and didn't really re-use data in a good way.